src/sqlfluff/dialects/dialect_sqlite.py

```diff
@@ -149,3 +431,42 @@ class TableConstraintSegment(ansi.TableConstraintSegment):
             optional=True,
         ),
     )
+
+
+class TransactionStatementSegment(ansi.TransactionStatementSegment):
+    """A `COMMIT`, `ROLLBACK` or `TRANSACTION` statement.
+
+    As per https://www.sqlite.org/lang_transaction.html
+    """
+
+    type = "transaction_statement"
+    match_grammar: Matchable = Sequence(
+        OneOf("BEGIN", "COMMIT", "ROLLBACK", "END"),
+        OneOf("TRANSACTION", optional=True),
+        Sequence("TO", "SAVEPOINT", Ref("ObjectReferenceSegment"), optional=True),
+    )
+
+
+class StatementSegment(ansi.StatementSegment):
+    """Overriding StatementSegment to allow for additional segment parsing."""
+
+    match_grammar = ansi.StatementSegment.match_grammar
+
+    parse_grammar: Matchable = OneOf(
+        Ref("AlterTableStatementSegment"),
+        Ref("CreateIndexStatementSegment"),
+        Ref("CreateTableStatementSegment"),
+        Ref("CreateTriggerStatementSegment"),
+        Ref("CreateViewStatementSegment"),
+        Ref("DeleteStatementSegment"),
+        Ref("DropIndexStatementSegment"),
+        Ref("DropTableStatementSegment"),
+        Ref("DropTriggerStatementSegment"),
+        Ref("DropViewStatementSegment"),
+        Ref("ExplainStatementSegment"),
+        Ref("InsertStatementSegment"),
+        Ref("SelectableGrammar"),
+        Ref("TransactionStatementSegment"),
+        Ref("UpdateStatementSegment"),
+        Bracketed(Ref("StatementSegment")),
+    )
```

Comment:
For this grammar, because most is a copy of the root one, but with some things removed - you could use the `.copy()` method to avoid repeating all the duplicated ones here, and just remove the ones you don't need.

Something like:
```python
parse_grammar = ansi.StatementSegment.parse_grammar.copy(remove=...)
```

Given just how _many_ you're removing - I can see that it might actually be _less_ duplication to keep things how you've done it already - but worth considering.

Reply:
I agree in principle, but like you've said, I'm removing many. This also affords sqlite some extra regression protection by stopping it getting anything we may add to ansi in the future. It's very very lightweight compared to every other dialect, which is why I think being explicit is cleaner.